XdmfWriter.h

```diff
@@ -64,7 +65,7 @@ namespace xdmfwriter
 /**
  * Writes data in XDMF format
  */
-template<TopoType Topo, typename T>
+template<TopoType Topo, typename T, typename D>
```

Comment:
Maybe you can also call these `VertexDataType` and `CellDataType` to make it more verbose.

Reply:
You could also write typename D = T in order to not break existing code.

Reply:
Thanks! I like both suggestions!